src/serializer/ResidualHeapVisitor.js

```diff
@@ -383,8 +394,35 @@ export class ResidualHeapVisitor {
         if (functionInfo.modified.has(innerName)) residualFunctionBinding.modified = true;
       }
     });
-
+    let useClassProperties = false;
+    let isConstructor = false;
+    let isComputed = val.$HasComputedName;
+
+    if (val.$HomeObject instanceof AbstractValue) {
```

Comment:
The Flow-type of `$HomeObject` doesn't allow for that. It's either `undefined`, or an `ObjectValue`. You should assert that here, and simplify the control-flow.